src/stan/model/indexing/lvalue.hpp

```diff
@@ -129,7 +130,8 @@ inline void assign(Vec1&& x,
  * the indexed size.
  */
 template <typename Vec1, typename Vec2,
-          require_all_eigen_vector_t<Vec1, Vec2>* = nullptr>
+          require_all_vector_t<Vec1, Vec2>* = nullptr,
+          require_all_not_std_vector_t<Vec1, Vec2>* = nullptr>
```

Comment:
`require_all_vector_t` is for any vector (Eigen row and col vectors, `var<row/col vector>`, and `std::vector<>`). We need the `not_std_vector_t<>` part here because we don't have anything in the library for just eigen vector or var vector